### DISCUSSION TITLE
Raycast Hoarder Extension

### DISCUSSION THREAD

**Discussion by kdurek** [karakeep-app/karakeep#370] [Show and tell]
Hey, if someone using Raycast, I have made extension for Hoarder, for now it's simple search, but I will be updating it to provide more functionality. Full REST API in Hoarder would be great! @MohamedBassem :D

![image](https://github.com/user-attachments/assets/90c18f92-5908-4cca-a736-e5ce21c83dbe)

Here is the link: https://www.raycast.com/kdurek/hoarder

**Comment by huyz** [comment]
Sweet, thanks!

**Comment by MohamedBassem** [comment]
Hey, not a raycast user myself, but this looks pretty cool. Thanks for building it! Btw, hoarder stores the `favicon` of every website it crawls, which I think can make the extension look even cooler if you show the favicon of each website. Also, regarding the REST API, I see that the extension is written in javascript? I think it might be easier if I publish the tRPC package to npm. This gives you a type-safe API for hoarder APIs.

**Reply by kdurek** [reply]
Hey thanks for the tip with favicon! It was painful to get it working on trpc endpoint :D 
Also good idea with this npm package would be awesome if it worked. My only concern is about that raycast uses its own fetching hook "useFetch" we will see if it will break something or will just work

**Reply by MohamedBassem** [reply]
Hmmm, trpc allows you to override the fetch function (https://trpc.io/docs/client/links/httpLink) but I see it can be tricky if raycast only provide a react hook.

**Reply by kdurek** [reply]
yeah, If you have time to push the library will be pleasure to try try! :D

**Reply by kdurek** [reply]
@MohamedBassem bump

**Reply by kdurek** [reply]
@MohamedBassem I see REST API is there, but I am missing search functionality, or maybe I can't find it?

**Reply by MohamedBassem** [reply]
@kdurek Really sorry for the late response. I forgot the search API in the REST API. Will include it in the next release. Sorry.

**Reply by kdurek** [reply]
No problem, will wait! :D

**Reply by yinan-c** [reply]
Thanks for the cool stuff, I have also created an [alfredworkflow](https://github.com/yinan-c/alfred-hoarder) using trpc, the REST API for now can only fetch max of 100 items, it would be cool to have search functionality in the REST API.

PS; for now searching is only possible for links not for texts and images(assets). Was wondering if @MohamedBassem you would implement search for texts and images.

**Reply by MohamedBassem** [reply]
I'm sorry that this took that long but search is now available via the rest API at `GET /bookmarks/search` ([docs](https://docs.hoarder.app/api/search-bookmarks)). The new release also includes a typescript SDK in case it makes your life better!
